File: TODO/todo/app/src/main/java/com/example/todo_app/MainActivity.java
package com.example.todo_app;

import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.DatePicker;

import com.example.todo_app.ui.login.LoginActivity;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.auth.FirebaseAuth;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import android.widget.Toast;

import java.util.Calendar;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    public BottomNavigationView navView;//змінна навігації додатку

    private int mYear, mMonth, mDay; //числові змінні дати
    String date; //текстова змінна з датою

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        navView = findViewById(R.id.nav_view);

        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.navigation_home, R.id.navigation_dashboard, R.id.navigation_notifications)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(navView, navController);

    }
    //Метод навігації для переходу до наступного вікна
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home){
            this.onBackPressed();
            navView.setVisibility(navView.VISIBLE);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
    //Метод виклику діалогу для виходу з профіля користувача
    private void openSignOutDialog(Context context) {
        AlertDialog.Builder quitDialog = new AlertDialog.Builder(
                this);
        quitDialog.setTitle("Вы уверены что хотите выйти из вашего акаунта?");

        quitDialog.setPositiveButton("Да", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                FirebaseAuth.getInstance().signOut();
                Intent intent = new Intent(context,LoginActivity.class);
                startActivity(intent);
            }
        });

        quitDialog.setNegativeButton("Нет", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
            }
        });

        quitDialog.show();
    }
    //Метод який виконує пошук заявок за датою
    private void DateDia(View view){
        final Calendar c = Calendar.getInstance();
        mYear = c.get(Calendar.YEAR);
        mMonth = c.get(Calendar.MONTH);
        mDay = c.get(Calendar.DAY_OF_MONTH);
        DatePickerDialog dpd = new DatePickerDialog(this,
                new DatePickerDialog.OnDateSetListener() {
                    @Override
                    public void onDateSet(DatePicker view, int year,
                                          int monthOfYear, int dayOfMonth) {
                        if ((dayOfMonth<10)&&((monthOfYear+1)<10))
                            date = ("0" + dayOfMonth + "." + "0" + (monthOfYear+1) + "." + year);
                        else if((monthOfYear+1)<10)
                            date = (dayOfMonth + "." + "0" + (monthOfYear+1) + "." + year);
                        else if (dayOfMonth<10)
                            date = ("0" + dayOfMonth + "." + (monthOfYear+1) + "." + year);
                        else
                            date = (dayOfMonth + "." + (monthOfYear+1) + "." + year);
                        Intent intent = new Intent(view.getContext(), DataFindActivity.class);
                        intent.putExtra("finddate",date);
                        startActivity(intent);
                    }
                }, mYear, mMonth, mDay);
        dpd.show();
        Toast.makeText(view.getContext(),"Оберіть дату",Toast.LENGTH_LONG).show();

    }

    //Метод натиску на кнопки
    @Override
    public void onClick(View view) {
        Intent intentadd = new Intent(view.getContext(),AddTaskActivity.class);
        Intent intentpro = new Intent(view.getContext(),ProfileActivity.class);
        switch (view.getId()){
            case R.id.floatingActionButton: startActivity(intentadd);break;
            case R.id.floatingActionButton2:startActivity(intentadd);break;
            case R.id.floatingActionButton3:DateDia(view);break;
            case R.id.floatingActionButton5:DateDia(view);break;
            case R.id.butt_singOut:openSignOutDialog(this);break;
            case R.id.button_profile:startActivity(intentpro);break;
            case R.id.button_about:openAboutDialog();break;
        }
    }
    //Метод виклику діалогу "о програмі"
    private void openAboutDialog() {
        AlertDialog.Builder quitDialog = new AlertDialog.Builder(
                this);
        quitDialog.setTitle(getResources().getString(R.string.about));
        quitDialog.setMessage(getResources().getString(R.string.about_text));

        quitDialog.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
            }
        });
        quitDialog.show();
    }
}

File: Calendar/Calendar/app/src/main/java/com/example/calendar/MainActivity.kt
package com.example.calendar

import android.app.DatePickerDialog
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.EditText
import android.widget.ImageButton
import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.room.Room
import com.example.calendar.database.MyDB
import com.google.android.material.floatingactionbutton.FloatingActionButton
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale


class MainActivity : AppCompatActivity(), AdapterView.AdapterClickListener {

    private lateinit var  btnAdd : FloatingActionButton
    private lateinit var db : MyDB
    private lateinit var search : EditText
    private lateinit var search_but : ImageButton
    private lateinit var data : ImageButton
    private var date = SimpleDateFormat("dd.MM.yyyy").format(System.currentTimeMillis())
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnAdd= findViewById(R.id.add)
        search= findViewById(R.id.edit_search)
        search_but= findViewById(R.id.search)
        data= findViewById(R.id.data)
        btnAdd.setOnClickListener{
            onClick()
        }
        db = MyDB.get(this)
        val recyclerView = findViewById<RecyclerView>(R.id.lista)
        val adapter = AdapterView(emptyList())
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)

        db.EventDAO().getAll().observe(this, { events ->
            adapter.events = events
            adapter.notifyDataSetChanged()
        })
        adapter.addOnClickListener(this)
        search_but.setOnClickListener {
            if(!search.text.isEmpty()){
                lifecycleScope.launch {
                    val db = MyDB.get(applicationContext)
                    val events = withContext(Dispatchers.IO) {
                        db.EventDAO().searchTitle("%"+search.text.toString()+"%")
                    }
                    adapter.events = events
                    adapter.notifyDataSetChanged()
                }
            }
        }

        var cal = Calendar.getInstance()
        var myFormat = "dd.MM.yyyy"
        var sdf = SimpleDateFormat(myFormat, Locale.US)

        val dateSetListener = DatePickerDialog.OnDateSetListener { view, year, monthOfYear, dayOfMonth ->
            cal.set(Calendar.YEAR, year)
            cal.set(Calendar.MONTH, monthOfYear)
            cal.set(Calendar.DAY_OF_MONTH, dayOfMonth)
            date = sdf.format(cal.time)

                lifecycleScope.launch {
                    val db = MyDB.get(applicationContext)
                    val events = withContext(Dispatchers.IO) {
                        db.EventDAO().searchDate("%"+date+"%")
                    }
                    adapter.events = events
                    adapter.notifyDataSetChanged()
                }

        }
        data.setOnClickListener {
            DatePickerDialog(this, dateSetListener,
                cal.get(Calendar.YEAR),
                cal.get(Calendar.MONTH),
                cal.get(Calendar.DAY_OF_MONTH)).show()
        }

        search.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable?) {
                val searchTerm = s.toString()
                if (searchTerm.isEmpty()) {
                    lifecycleScope.launch {
                        db.EventDAO().getAll().observe(this@MainActivity, { events ->
                            adapter.events = events
                            adapter.notifyDataSetChanged()

                        })
                    }
                }
            }
        })
    }

    fun onClick(){
        this.finish()
        val addIntent = Intent(this, NewEvent::class.java)
        startActivity(addIntent)
    }

    override fun onClick(id: Long) {
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Usunięcie wydarzenia")
        builder.setMessage("Usunąć zapis?")
        builder.setPositiveButton("Tak") { dialog, which ->
            lifecycleScope.launch {
                withContext(Dispatchers.IO) {
                    val db = MyDB.get(applicationContext)
                    db.EventDAO().delete(id)
                }
            }
        }
        builder.setNegativeButton("Nie") { dialog, which ->
        }
        val dialog = builder.create()
        dialog.show()
    }

}